### PR TITLE
Add Filename.null

### PR DESCRIPTION
**Pull request by nojb**
This PR proposes to add `Filename.null`, the name of the "null" device in each supported system.

### CODE REVIEW

File: stdlib/filename.mli
Comment (xavierleroy) [inline stdlib/filename.mli]: I would not put this as the first export in `filename.mli`.  This is not what users of the `Filename` module are looking for with highest priority. 

What about exporting and documenting it between `dirname` and `temp_file`?  `temp_file` is about generating file names for output, and `null` is also a file name for output, to some extent.

**Review by xavierleroy — CHANGES_REQUESTED** [review]
Nice idea.  I'm in favor, except for the minor point below.  

I tried to think of a more descriptive name than `null` and failed.

File: stdlib/filename.mli
Comment (nojb) [reply stdlib/filename.mli]: Indeed; I amended as suggested. Thanks!

Comment (nojb) [reply stdlib/filename.mli]: If you are happy with the state of the PR, could you formally "accept" it? Thanks!

**Review by xavierleroy — APPROVED** [review]
I am happy with the changes.  However, I agreed a while ago to not review stdlib changes and leave this to an ad-hoc "stdlib subcommitee", so you need one more approval from someone who is more involved in stdlib design and evolutions.

**Comment by alainfrisch** [comment]
This is a nice addition, also aligned with https://github.com/dbuenzli/bos/blob/61526bf67f218fd730db8ac19149945eb9c97d20/src/bos.mli#L779

May I just suggest to take inspiration from the docstring above, which is more explicit than the current one?

**Comment by nojb** [comment]
> May I just suggest to take inspiration from the docstring above, which is more explicit than the current one?

Thanks for the pointer, I amended as suggested.

**Review by alainfrisch — APPROVED** [review]
LGTM.  I think we can live without a dedicated test, but if you feel like adding one (e.g. checking that reading from `null` indeed raises `End_of_file`), don't hesitate.  Otherwise, I'll merge in a few days.

**Comment by nojb** [comment]
> LGTM. I think we can live without a dedicated test, but if you feel like adding one (e.g. checking that reading from `null` indeed raises `End_of_file`), don't hesitate. Otherwise, I'll merge in a few days.

Added a test as suggested, thanks.

**Pull request merged**
